### ISSUE TITLE
Call to undefined method League\Flysystem\Filesystem::getAdapter()

### ISSUE THREAD

**Issue by ShamarKellman** [laravel-filament/filament#1464]
### Package

filament/spatie-laravel-media-library-plugin

### Package Version

v2.9.19

### Laravel Version

v9.0.0-beta.5

### Livewire Version

v2.10.1

### PHP Version

PHP 8.1.1

### Bug description

Laravel 9 uses Flysystem v3 which removes the `getAdapter()` method used in the [SpatieMediaLibraryFileUpload.php - line 66](https://github.com/laravel-filament/filament/blob/f534cb030a2055787b7f6e452a8440fd002ae1ef/packages/spatie-laravel-media-library-plugin/src/Forms/Components/SpatieMediaLibraryFileUpload.php#L66)

**Comment by danharrin** [comment]
So what is the alternative in Flysystem 3?

**Comment by ShamarKellman** [comment]
Found a comment on the repo regarding the removal of the method [https://github.com/thephpleague/flysystem/issues/1142#issuecomment-596671929](https://github.com/thephpleague/flysystem/issues/1142#issuecomment-596671929)

**Comment by ShamarKellman** [comment]
After some playing around I found something that might work 
```php 
$this->getUploadedFileUrlUsing(function (SpatieMediaLibraryFileUpload $component, string $file): ?string {
            /** @var FilesystemAdapter $storage */
            $storage = $component->getDisk();

            if (! $component->getRecord()) {
                return null;
            }

            /** @var ?Media $media */
            $media = Media::findByUuid($file);

            if (
                $storage->getAdapter() instanceof AwsS3Adapter &&
                $this->getVisibility() === 'private'
            ) {
                return $media?->getTemporaryUrl(now()->addMinutes(5));
            }

            return $media?->getUrl();
        });
 ```

So we can call getAdapter() on the `$storage` variable instead of using `$storageDriver` 

Hopefully this helps

**Comment by danharrin** [comment]
The `$storage->getAdapter()` method doesn't exist for Laravel 8 though :( So yes it would fix for L9, but is not backwards compatible.

**Comment by danharrin** [comment]
I couldn't think of a reason we needed this check in the first place, so for simplicity I have removed it. Let me know if it messes anything up in your app. The fix will be available in v2.10.0 when it is released.

**Comment by AhmadRaza-007** [comment]
copy this and paste it in the vendor/league/flysystem/src/Filesystem.php file. hope this will work and will not create any problem.
